core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -104,7 +105,8 @@ protected final SdkClientConfiguration mergeChildDefaults(SdkClientConfiguration
                                   .option(AwsAdvancedClientOption.ENABLE_DEFAULT_REGION_DETECTION, true)
                                   .option(AwsClientOption.CREDENTIALS_PROVIDER, DefaultCredentialsProvider.create())
                                   .option(SdkClientOption.RETRY_POLICY, AwsRetryPolicy.defaultRetryPolicy())
-                                  .option(AwsClientOption.SERVICE_SIGNING_NAME, signingName()));
+                                  .option(AwsClientOption.SERVICE_SIGNING_NAME, signingName())
+                                  .option(SdkAdvancedClientOption.DISABLE_HOST_PREFIX_INJECTION, false));
```

Comment:
Why is this an option?

Reply:
IMO it is an advanced option and not common use case to set it, so I made it an option instead of directing exposing it in [ClientOverrideConfiguration](https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java). Do you think it makes more sense to expose it as top level member in ClientOverrideConfiguration?

Reply:
Can it not be an option at all?

Reply:
Talked offline. It can be useful in testing.